warehouse/oidc/tasks.py

```diff
@@ -57,3 +58,21 @@ def compute_oidc_metrics(request):
         "warehouse.oidc.total_critical_projects_published_with_oidc_publishers",
         projects_published_with_oidc.where(Project.pypi_mandates_2fa.is_(True)).count(),
     )
+
+    # Metric for total number of files published via OIDC
+    metrics.gauge(
+        "warehouse.oidc.total_files_published_with_oidc_publishers",
+        request.db.query(File.Event)
+        .where(File.Event.additional.op("->>")("publisher_url").is_not(None))
+        .count(),
+    )
+
+    # Number of publishers for specific publishers
+    for t in request.db.query(OIDCPublisher.discriminator).distinct().all():
+        discriminator = t[0]
+        metrics.gauge(
+            f"warehouse.oidc.{discriminator}.publishers",
```

Comment:
Question: is the discriminator better provided as a tag instead of the name, so we can sum by all, and split by discriminator?